sdk/angelscript/projects/cmake/CMakeLists.txt

```diff
@@ -178,37 +178,38 @@ install(TARGETS ${ANGELSCRIPT_LIBRARY_NAME} EXPORT AngelscriptTargets
 
 install(FILES
 	${CMAKE_CURRENT_SOURCE_DIR}/../../include/angelscript.h
-	DESTINATION include
+	DESTINATION include/${ANGELSCRIPT_LIBRARY_NAME}
 	COMPONENT Devel
 )
 
+
 include(CMakePackageConfigHelpers)
 write_basic_package_version_file(
-	"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptConfigVersion.cmake"
+	"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/${ANGELSCRIPT_LIBRARY_NAME}ConfigVersion.cmake"
 	VERSION ${PROJECT_VERSION}
 	COMPATIBILITY AnyNewerVersion
 )
 
 export(EXPORT AngelscriptTargets
-	FILE "${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptTargets.cmake"
-	NAMESPACE Angelscript::
+	FILE "${CMAKE_CURRENT_BINARY_DIR}/Angelscript/${ANGELSCRIPT_LIBRARY_NAME}Targets.cmake"
+	NAMESPACE AngelScript::
 )
-configure_file(cmake/AngelscriptConfig.cmake
-	"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptConfig.cmake"
-	COPYONLY
+
+set(TARGETS_EXPORT_NAME "${ANGELSCRIPT_LIBRARY_NAME}Targets")
+configure_package_config_file(
+    cmake/Config.cmake.in
+    "${CMAKE_CURRENT_BINARY_DIR}/Angelscript/${ANGELSCRIPT_LIBRARY_NAME}Config.cmake"
+    INSTALL_DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/Angelscript/${ANGELSCRIPT_LIBRARY_NAME}Config.cmake"
```

Comment:
That is wrong.

Reply:
That's the only way I could make the thing pass the tests and place files in the correct directories. CMake documentation on what pathes are expected, whether relative or absolute, relative to what - is completely unclear here. The code above works and compiles and finds the library, the code from original file would not.

Reply:
> That's the only way I could make the thing pass the tests and place files in the correct directories.

It works accidentally, the problem is somewhere else. Current value doesn't making any sense at all.

> CMake documentation on what pathes are expected, whether relative or absolute, relative to what - is completely unclear here.

Absolute is the path with `CMAKE_INSTALL_PREFIX` which is usually not a recommended way. The relative path is relative to install directory specified by the user. As with the rest of the destinations here.

Examples:
- https://github.com/forexample/package-example/blob/master/Foo/CMakeLists.txt#L163-L171
- https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#example-generating-package-files

Reply:
Think I tried it and it always silently failed (which I hate about this kind of stuff, I literally have no sign what's wrong here, just nothing happens). When I did it in a way I thought it should work it would never create that file, when I went for something completely stupid like you see above it started to work. It's literally trial and error because I couldn't make it work the right way as I understood the function.